### PR TITLE
feat: add a new example

### PR DESCRIPTION
**Pull request by xXenvy**
This pull request introduces a new example to the lib. The purpose of this demo is to illustrate the automatic generation of setter and getter methods for private fields.

### CODE REVIEW

**Comment by JhnW** [comment]
Maybe change example name (folder) to getters _generated or, even better, accessor.

**Comment by JhnW** [comment]
Please check doc_src/demos/ folder and please add your own doc for demo. From what I remember, just adding a file is enough, but if necessary, check the local documentation generation.

File: examples/access_generator/main.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    def should_ignore_field(comment: Optional[Comment]) -> bool:
+        return "devana: ignore-field" in get_devana_comments(comment) if comment else False
+
+    for private_field in filter(lambda field: isinstance(field, FieldInfo), class_info.private):
+        if should_ignore_field(private_field.associated_comment):
+            continue
+
+        class_info.content.extend(create_field_methods(field=private_field))
+
+    return None
```
Comment (JhnW) [inline examples/access_generator/main.py:80]: When a function returns nothing, do not use an explicit return None.

File: examples/access_generator/main.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    field_getter.modification |= FunctionModification.CONST
+
+    if not should_ignore_attributes(field.associated_comment):
+        field_getter.attributes.append(AttributeDeclaration([nodiscard_attribute, maybe_unused_attribute]))
+        field_setter.attributes.append(AttributeDeclaration([maybe_unused_attribute]))
+
+    return field_setter, field_getter
+
+
+def create_class_methods(class_info: ClassInfo) -> None:
```
Comment (JhnW) [inline examples/access_generator/main.py:69]: When a function returns nothing, it is customary to not use None as a hint about the return type.  Please, remove the annotations.
For the future, None is used for something else, such as a union or an optional. To describe functions that return nothing uses:
> from typing import NoReturn
> def foop() -> NoReturn:

File: examples/access_generator/main.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        header_file: SourceFile = SourceFile()
+        header_file.type = SourceFileType.HEADER
+        header_file.header_guard = file_name.upper().replace(".", "_")
+        header_file.includes.append(iostream_include)
+
+        for class_info in file_content:
+            create_class_methods(class_info)
+            header_file.content.append(class_info)
+
+        with open(f"./output/{file_name}", "w+") as f:
```
Comment (JhnW) [inline examples/access_generator/main.py:110]: You should define encoding for open. Also constructor in generated code is broken.

**Comment by JhnW** [comment]
Please check why output does not generate comments with correct spacing.

**Review by JhnW — CHANGES_REQUESTED** [review]

File: examples/access_generator/main.py (hunk 2 above)
Comment (xXenvy) [reply examples/access_generator/main.py:69]: Z tego co wiem, to `NoReturn` jest stosowane tylko w sytuacjach, gdy funkcja faktycznie nic nie zwraca (np. Wywala wyjątek). Takie przypadki są rzadkie, a funkcje domyślnie zwracają `NoneType`.

**Comment by xXenvy** [comment]
> Please check doc_src/demos/ folder and please add your own doc for demo. From what I remember, just adding a file is enough, but if necessary, check the local documentation generation.

Done.

**Comment by xXenvy** [comment]
> Maybe change example name (folder) to getters _generated or, even better, accessor.

I changed to `accessor_generator`. In my opinion, it is ok.

File: examples/access_generator/main.py (hunk 1 above)
Comment (xXenvy) [reply examples/access_generator/main.py]: Done

File: examples/access_generator/main.py (hunk 2 above)
Comment (xXenvy) [reply examples/access_generator/main.py]: As far as I know, `NoReturn` is only used in situations where the function actually returns nothing (e.g., it throws an exception). Such cases are rare, and functions return `NoneType` by default, in this case also.

Comment (JhnW) [reply examples/access_generator/main.py]: Yes, you're right - it was a side note before such hints.
We don't want None anyway: https://pylint.readthedocs.io/en/latest/user_guide/messages/refactor/useless-return.html

Comment (xXenvy) [reply examples/access_generator/main.py]: Done in https://github.com/JhnW/devana/pull/16/commits/6cb2adf143e551c5d92e1a76e853e6d2389be2c3

File: examples/access_generator/main.py (hunk 3 above)
Comment (xXenvy) [reply examples/access_generator/main.py]: Done in https://github.com/JhnW/devana/pull/16/commits/6cb2adf143e551c5d92e1a76e853e6d2389be2c3

File: examples/accessors_generator/input/car.h [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+class Car {
+private:
+    std::string brand;
+    float speed;
+
+    // devana: ignore-attributes
+    bool working;
+public:
+    [[maybe_unused]]
```
Comment (JhnW) [inline examples/accessors_generator/input/car.h:15]: It is not necessary.

File: examples/accessors_generator/main.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+            constructor.body = ""
+
+    for private_field in filter(lambda field: isinstance(field, FieldInfo), class_info.private):
+        if should_ignore_field(private_field.associated_comment):
+            continue
+
+        class_info.content.extend(create_field_methods(field=private_field))
+
+
+def load_files_content() -> Iterable[Tuple[str, Iterable[ClassInfo]]]:
```
Comment (JhnW) [inline examples/accessors_generator/main.py:90]: Please make from this named tuple or even better  dataclass.
> Iterable[Tuple[str, Iterable[ClassInfo]]]

It is the same horrible code in python like in c++ templates:

> Iterable<Tuple<str, Iterable<ClassInfo>>>>

**Pull request merged**